app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -186,6 +128,48 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
     association.push(new_records)
   end
 
+  def save_inventory_multi_child(rel_type, association, parent, hashes, find_key, child_keys = [], extra_keys = [],
```

Comment:
Can 2-3 methods be extracted out of this one?
This way it will be much clearer what is going on.